### ISSUE TITLE
Interface for a (functional) minibatch generator.

### ISSUE THREAD

**Issue by fkdosilovic** [rasbt/mlxtend#682]
I've noticed that in order to create minibatches from training set, the easiest what to achieve this, using mlxtend, is to inherit _IterativeModel and then use _yield_minibatches_idx. Have you thought about providing a more functional interface, i.e. providing a function (generator)?

Then the call would look something like this:
```python
for X_mb, y_mb in minibatch_gen(X_train, y_train):
    ...
```

**Comment by rasbt** [comment]
Good point, this was originally a function -- I used it when working with TensorFlow back in the day. However, within mlxtend, I didn't see the need for having it outside the model due to the sklearn-like API. Perhaps, it would be a useful thing to have for `partial_fit` supporting models though.

**Comment by jbbqqf** [comment]
Hi! Triaging older issues this week — this is a ~5-year-old feature request with no follow-up.

For the functional-minibatch use case, today's reasonable workaround is one of:

- `mlxtend.preprocessing.shuffle_arrays_unison` to shuffle, then a small loop — see [`mlxtend/preprocessing/shuffle.py`](https://github.com/rasbt/mlxtend/blob/master/mlxtend/preprocessing/shuffle.py).
- `sklearn.utils.gen_batches` or `gen_even_slices` for the index generation.
- For PyTorch users, `torch.utils.data.DataLoader`.

The underlying `_yield_minibatches_idx` referenced in the original post still lives on the protected `_IterativeModel` mixin (no public `minibatch_gen(X, y)` helper was ever exposed), and there's no open PR in flight.

Given the available workarounds and the years of silence, this can probably be closed — happy to reopen if a concrete PR materializes.

(Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the code references above were verified manually against the current `master`.)
